web/client/components/style/vector/Text.jsx

```diff
@@ -194,8 +195,37 @@ class Text extends React.Component {
                     />
                 </Col>
             </Row>
+            <Row>
+                <Col xs={6}>
+                    <Message msgId="draw.textRotation"/>
+                </Col>
+                <Col xs={6} style={{position: "static"}}>
+                    <div className="mapstore-slider with-tooltip">
+                        <Slider
+                            tooltips
+                            step={2}
+                            start={[style.textRotationDeg || 0]}
+                            format={{
+                                from: value => Math.round(parseFloat(value)),
+                                to: value => Math.round(value) + ' &deg;'
+                            }}
+                            range={{
+                                min: 0,
+                                max: 359
+                            }}
+                            onChange={(values) => {
+                                const rotationDeg = parseInt(values[0].replace(' &deg;', ''), 10);
+                                this.props.onChange(style.id, {textRotation: this.toRad(rotationDeg), textRotationDeg: rotationDeg});
```

Comment:
textRotation is a derived data so i would only add the deg version in the style with the needed changes to the part of the code